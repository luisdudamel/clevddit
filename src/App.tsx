import { Route, Routes } from "react-router-dom";
import Header from "./components/Header/Header";
import HomePage from "./pages/HomePage/HomePage";
import LoginRegisterPage from "./pages/LoginRegisterPage/LoginRegisterPage";
import PostPage from "./pages/PostPage/PostPage";
import "./styles/sass/styles.scss";

const App = (): JSX.Element => {
  return (
    <>
      <Routes>
        <Route path="/*" element={<LoginRegisterPage />}></Route>
        <Route path="/login" element={<LoginRegisterPage />}></Route>
        <Route path="/register" element={<LoginRegisterPage />}></Route>
        <Route
          path="/home"
          element={
            <>
              <Header />
              <HomePage />
            </>
          }
        ></Route>
        <Route
          path="/post/:postId"
          element={
            <>
              <Header />
              <PostPage />
            </>
          }
        ></Route>
      </Routes>
    </>
  );
};

export default App;
