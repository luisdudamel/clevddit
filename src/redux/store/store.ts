import { configureStore } from "@reduxjs/toolkit";
import postsSliceReducer from "../feature/postSlice/postsSlice";
import uiSliceReducer from "../feature/uiSlice/uiSlice";
import userSliceReducer from "../feature/userSlice/userSlice";
import usersSliceReducer from "../feature/usersSlice/usersSlice";

export const store = configureStore({
  reducer: {
    ui: uiSliceReducer,
    posts: postsSliceReducer,
    users: usersSliceReducer,
    user: userSliceReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
