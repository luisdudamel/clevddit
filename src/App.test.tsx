import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import TestRenderer from "react-test-renderer";
import App from "./App";
import { store } from "./redux/store/store";

describe("Given an App function", () => {
  describe("When it's called", () => {
    test("Then it should always match this snapshot", () => {
      const testedRegisterPage = TestRenderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <App />
          </BrowserRouter>
        </Provider>
      );

      expect(testedRegisterPage).toMatchSnapshot();
    });
  });
});
