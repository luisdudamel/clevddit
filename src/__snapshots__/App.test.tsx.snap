// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an App function When it's called Then it should always match this snapshot 1`] = `
<main
  className="login-page"
>
  <h1
    className="login-page__title"
  >
    Welcome to clevddit!
  </h1>
  <div
    className="login-form__container"
  >
    <div
      className="logo-container"
    >
      <img
        alt="Cleverpy company logo"
        className="login-form--logo"
        src="/img/cleverpy-logo.png"
      />
    </div>
    <form
      action="submit"
      className="login-form"
      onSubmit={[Function]}
    >
      <label
        hidden={true}
        htmlFor="username"
        id="username"
      >
        Username
      </label>
      <input
        aria-label="username"
        className="login-form--input"
        id="username"
        name="username"
        onChange={[Function]}
        placeholder="Username"
        type="text"
        value=""
      />
      <label
        hidden={true}
        htmlFor="password"
        id="password"
      >
        Password
      </label>
      <input
        aria-label="password"
        autoComplete="on"
        className="login-form--input"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
      <button
        className="login-register-button"
        disabled={true}
        type="submit"
      >
        Register
      </button>
    </form>
  </div>
  <div
    className="login-page__links__container"
  >
    <p>
      Already
       a member?
    </p>
    <p>
      <a
        className="login-page__links__text"
        href="/login"
        onClick={[Function]}
      >
        Go to login
      </a>
    </p>
  </div>
</main>
`;
